### DISCUSSION TITLE
DITA 2.0 adoption, DITA 1.x only features, and phasing in preprocess2

### DISCUSSION THREAD

**Discussion by jomjohn** [dita-ot/dita-ot#4146] [Q&A]
My team and I are working through some internal publishing requirements and I am hoping to gain a better understanding of what to expect as the DITA-OT gets more awesome. 

- In general, what are the thoughts around support for DITA 1.x features that are being dropped in DITA 2.0? For example, would you expect regressions around copy-to be addressed in future versions of the DITA-OT? When DITA 2.0 resourceid-based file renaming is supported in the DITA-OT will it co-exist with DITA 1.x copy-to?
- If preprocess2 issues are addressed in DITA-OT 4.x, what is the likelihood they would also be patched for DITA-OT 3.x?
- Is there a target milestone when preprocess2 will become the default for all bundled plugins and phase out preprocess (current)?
- Is there going to be a point in time where the DITA-OT will explicitly drop support for some DITA 1.x only features?

Thank you for any clarification that you can provide.

**Comment by chrispy-snps** [comment]
Hi @jomjohn,

I can share some thoughts/guesses.

My group does not use the DITA 1.x features that are being deprecated, so we have no particular thoughts about them. The chunking always seemed complicated and we avoided it. In fact, we prefer to author topics with subtopics directly inside them:

```
<topic id="root">
  <title>Features</title>
  <body>...</body>

  <topic id="featureA">
    <title>Feature A</title>
    <body>...</body>
  </topic>

  <topic id="featureB">
    <title>Feature B</title>
    <body>...</body>
  </topic>

</topic>
```

I do not believe that any 4.x fixes (in preprocess2 or anything else) will be backported to any 3.x releases. As far as I know, DITA-OT release development moves only in a forward release direction.

There is always interest in moving to preprocess2 as a default. :)  There is no fixed release or date target, but there was recent discussion about possibly giving it a bit more priority:

https://github.com/dita-ot/dita-ot/wiki/Meeting-minutes-2023-02-09#item-2-dita-ot-41-development-status-and-updates

I don't know the answer to the last question.

**Reply by jomjohn** [reply]
Thank you for your thoughts/guesses @chrispy-snps !

It makes sense why the dita-ot would just move forward and not patch older releases. In this case (between dita-ot 3.x and 4.x) where the prerequisite is Java version 17 for 4.x. Some environments may not be able to upgrade java for various reasons hence not being able to upgrade to the latest dita-ot. 

I agree to just focus on the latest and make it the greatest.

**Comment by jelovirt** [comment]
> * If preprocess2 issues are addressed in DITA-OT 4.x, what is the likelihood they would also be patched for DITA-OT 3.x?

Very unlikely because there are so few people working on DITA-OT.

> * Is there a target milestone when preprocess2 will become the default for all bundled plugins and phase out preprocess (current)?

No.

> * Is there going to be a point in time where the DITA-OT will explicitly drop support for some DITA 1.x only features?

Maybe, we don't know. We want to keep supporting DITA 1.x even after DITA 2.x is released. The only real reason to drop some 1.x only feature is that maintaining it is too time consuming for the team in comparison to how useful/used it is.
